internal/services/oracle/oracle_autonomous_database_clone_resource.go

```diff
@@ -0,0 +1,683 @@
+			if err != nil {
+				return err
+			}
+
+			if err := client.DeleteThenPoll(ctx, *id); err != nil {
+				return fmt.Errorf("deleting %s: %+v", *id, err)
+			}
+
+			return nil
+		},
+	}
+}
+
+func (r AutonomousDatabaseCloneResource) Update() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			// nothing to update
+			return nil
+		},
+	}
+}
+
+func (r AutonomousDatabaseCloneResource) IDValidationFunc() pluginsdk.SchemaValidateFunc {
+	return autonomousdatabases.ValidateAutonomousDatabaseID
+}
+
+func expandCloneCustomerContacts(input []string) []autonomousdatabases.CustomerContact {
+	if len(input) == 0 {
+		return nil
+	}
+
+	contacts := make([]autonomousdatabases.CustomerContact, 0, len(input))
+	for _, email := range input {
+		if strings.TrimSpace(email) != "" {
+			contacts = append(contacts, autonomousdatabases.CustomerContact{
+				Email: email,
+			})
+		}
+	}
+	return contacts
+}
+
+func expandCloneCustomerContactsPtr(input []string) *[]autonomousdatabases.CustomerContact {
+	if len(input) == 0 {
+		return nil
+	}
+
+	contacts := expandCloneCustomerContacts(input)
+	return &contacts
+}
+
+func flattenCloneCustomerContacts(input []autonomousdatabases.CustomerContact) []string {
+	if len(input) == 0 {
+		return nil
+	}
+
+	emails := make([]string, 0, len(input))
+	for _, contact := range input {
+		if contact.Email != "" {
+			emails = append(emails, contact.Email)
+		}
+	}
+	return emails
+}
+
+func (AutonomousDatabaseCloneResource) CustomizeDiff() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Second,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			if metadata.ResourceData == nil {
+				return nil
+			}
+
+			sourceId := metadata.ResourceData.Get("source_id").(string)
+			dbWorkload := metadata.ResourceData.Get("db_workload").(string)
+
+			if sourceId == "" || dbWorkload == "" {
+				return nil
+			}
+
+			if metadata.ResourceData.Id() != "" {
+				return nil
+			}
+
+			sourceWorkload, err := getSourceWorkload(ctx, sourceId, metadata)
+			if err != nil {
+				return err
+			}
+
+			targets, exists := workloadMatrix[sourceWorkload]
+			if !exists {
+				return fmt.Errorf("unsupported source workload: %s", sourceWorkload)
+			}
+
+			for _, target := range targets {
+				if dbWorkload == target {
+					return nil
+				}
+			}
+
+			return fmt.Errorf("invalid workload: %s->%s not allowed", sourceWorkload, dbWorkload)
+		},
+	}
+}
+
+var workloadMatrix = map[string][]string{
+	"DW":   {"OLTP", "DW"},
+	"OLTP": {"DW", "OLTP"},
+	"AJD":  {"OLTP", "DW", "APEX"},
+	"APEX": {"AJD", "OLTP", "DW"},
+}
+
+func getSourceWorkload(ctx context.Context, sourceId string, metadata sdk.ResourceMetaData) (string, error) {
+	id, err := autonomousdatabases.ParseAutonomousDatabaseID(sourceId)
+	if err != nil {
+		return "", fmt.Errorf("invalid source_id format: %v", err)
```

Comment:
We generally don't wrap the error messages returned by `Parse.*ID` functions

```suggestion
		return "", err
```